decidim/decidim
# Fix user related absolute URLs

#### :tophat: What? Why?
This is continuation for #9550.

During the review I noticed that the user related absolute URL were still incorrect in development where they did not have the correct port for the running server. This fixes that issue.

At the same time I noticed that the `default_avatar_url` method was broken so I fixed that too.

#### :pushpin: Related Issues
- Related to #9550

#### Testing
Following the use case at #9550:
- Development app with default seeds
- Login as the admin account and make sure you have the notification setting in your profile settings configured to "real time"
- In another browser session, login as `user@example.org`
- Authorize that user using the example authorization, type in `123456X` as the document number (other details up to you)
- Once authorized, log out
- Create a new user account, e.g. `another@example.org`
- Confirm the new user account (using `/letter_opener`) and login with it
- Authorize the new user account using the example authorization and type in `123456X` as the document number (other details up to you)
- See an authorization error
- Go to `/letter_opener` and now there should be a new admin notification regarding the authorization conflict
- Check that clicking the links in that email work in the local development environment (see the screenshot below)

### :camera: Screenshots
After this fix, these links should also work now in local development environment:

![The links in the email to check](https://user-images.githubusercontent.com/864340/180229060-ac818118-62b9-4f2c-a8d2-f8d46f253547.png)
